### PR TITLE
`Paywalls`: basic localization support

### PR DESCRIPTION
**Pull request by NachoSoto**
Most of the changes are to support the creation of the new `PaywallViewLocalizationTests` by being able to override the environment `Locale`.

Simulator Screenshot - iPhone 14 Pro - 2023-07-20 at 18 10 20

### CODE REVIEW

File: RevenueCatUI/Views/AsyncButton.swift [hunk 1]
```diff
@@ -47,7 +47,7 @@ struct AsyncButton<Label>: View where Label: View {
             Button {
                 self.error = nil
             } label: {
-                Text("OK")
+                Text("OK", bundle: .module)
```
Comment (joshdholtz) [inline RevenueCatUI/Views/AsyncButton.swift:50]: Does `.module` work with cocoapods? Asking this if we decided to try and call `RevenueCatUI` views from React Native, Flutter, etc 🤷‍♂️

File: RevenueCatUI/Data/Localizable.xcstrings
Comment (joshdholtz) [inline RevenueCatUI/Data/Localizable.xcstrings:1]: 😍

**Review by joshdholtz — APPROVED** [review]

File: RevenueCatUI/Views/AsyncButton.swift (hunk 1 above)
Comment (NachoSoto) [reply RevenueCatUI/Views/AsyncButton.swift:50]: Yes :) it’d be equivalent to `Bundle(forClass:)`

Comment (NachoSoto) [reply RevenueCatUI/Views/AsyncButton.swift:50]: But we'd have to manually add support for CocoaPods anyway, which we won't initially.

**Pull request merged**